### PR TITLE
Add support for grouping result using a function

### PR DESCRIPTION
**Pull request by pavelsmolka**
Add functionality that checks whether the result data in
ResponseParser\Component\Grouping contain groups created by
solr parameter `group.func` (an arbitrary Solr-supported
function).

When such field is found, apply the same parsing logic
which is usually used for field-grouping, in order to ensure
the function-grouping is recognized and added to the FieldGroup
result object.

This should add a new functionality requested in ticket #440

### CODE REVIEW

File: library/Solarium/QueryType/Select/ResponseParser/Component/Grouping.php [hunk 1]
```diff
@@ -67,7 +67,9 @@ public function parse($query, $grouping, $data)
             // parse field groups
             $valueResultClass = $grouping->getOption('resultvaluegroupclass');
             $documentClass = $query->getOption('documentclass');
-            foreach ($grouping->getFields() as $field) {
+
+            // check grouping fields as well as the grouping function (either can be used in the query)
+            foreach (array_merge($grouping->getFields(), [$grouping->getFunction()]) as $field) {
```
Comment (basdenooijer) [inline library/Solarium/QueryType/Select/ResponseParser/Component/Grouping.php]: Currently PHP 5.3 is still supported, so short array syntax must be avoided. The next major Solarium release will update the requirements.

Comment (basdenooijer) [inline library/Solarium/QueryType/Select/ResponseParser/Component/Grouping.php]: I just see the current CI config allows 5.3 failures, that's why the tests pass. But the composer file still requires 5.3.2 and I cannot change that in a minor release.

**Comment by basdenooijer** [comment]
Thanks for you fix, apart from one small issue regarding short array syntax the changes look good. Could you fix the array, and retarget the PR to the develop branch? Then I will merge it.

**Comment by pavelsmolka** [comment]
I have fixed the array syntax and rebased the branch - it should be ready to get merged now.

**Comment by basdenooijer** [comment]
Thanks!

**Pull request merged**